presto-hive/src/test/java/io/prestosql/plugin/hive/TestHiveIntegrationSmokeTest.java

```diff
@@ -853,6 +853,126 @@ public void testTableAuthorizationForRole()
         assertUpdate(admin, "DROP ROLE admin");
     }
 
+    @Test
+    public void testViewAuthorization()
+    {
+        Session admin = Session.builder(getQueryRunner().getDefaultSession())
+                .setCatalog(getSession().getCatalog().get())
+                .setIdentity(Identity.forUser("hive").withRole("hive", new SelectedRole(ROLE, Optional.of("admin"))).build())
+                .build();
+
+        Session alice = testSessionBuilder()
+                .setCatalog(getSession().getCatalog().get())
+                .setIdentity(Identity.forUser("alice").build())
+                .build();
+
+        assertUpdate(admin, "CREATE SCHEMA test_view_authorization");
+        assertUpdate(admin, "CREATE VIEW test_view_authorization.bar AS SELECT current_user AS user");
+
+        assertAccessDenied(
+                alice,
+                "ALTER VIEW test_view_authorization.bar SET AUTHORIZATION alice",
+                "Cannot set authorization for view test_view_authorization.bar to USER alice");
+        assertUpdate(admin, "ALTER VIEW test_view_authorization.bar SET AUTHORIZATION alice");
+        assertUpdate(alice, "ALTER VIEW test_view_authorization.bar SET AUTHORIZATION admin");
+
+        assertUpdate(admin, "DROP VIEW test_view_authorization.bar");
+        assertUpdate(admin, "DROP SCHEMA test_view_authorization");
+    }
+
+    @Test
+    public void testViewAuthorizationSecurityDefiner()
+    {
+        Session admin = Session.builder(getQueryRunner().getDefaultSession())
+                .setCatalog(getSession().getCatalog().get())
+                .setIdentity(Identity.forUser("hive").withRole("hive", new SelectedRole(ROLE, Optional.of("admin"))).build())
+                .build();
+
+        Session alice = testSessionBuilder()
+                .setCatalog(getSession().getCatalog().get())
+                .setIdentity(Identity.forUser("alice").build())
+                .build();
+
+        assertUpdate(admin, "CREATE SCHEMA test_view_authorization");
+        assertUpdate(admin, "CREATE TABLE test_view_authorization.foo (col int)");
+        assertUpdate(admin, "INSERT INTO test_view_authorization.foo VALUES (1)", 1);
+        assertUpdate(admin, "CREATE VIEW test_view_authorization.bar SECURITY DEFINER AS SELECT * from test_view_authorization.foo");
```

Comment:
this is really unreadable to cope with `bar` and `foo`. I need to check couple times what is what when trying to understand error messages below.